src/server/controllers/model-version-purchase.controller.ts

```diff
@@ -0,0 +1,24 @@
+import { PurchaseModelVersionInput } from '~/server/schema/model-version-purchase.schema';
+import { Context } from '~/server/createContext';
+import { TRPCError } from '@trpc/server';
+import { throwAuthorizationError, throwDbError } from '~/server/utils/errorHandling';
+import { purchaseModelVersion } from '~/server/services/model-version-purchase.service';
+
+export const purchaseHandler = async ({
+  input,
+  ctx,
+}: {
+  input: PurchaseModelVersionInput;
+  ctx: DeepNonNullable<Context>;
+}) => {
+  try {
+    if (input.userId !== ctx.user.id && !ctx.user.isModerator) {
```

Comment:
Should the condition be this way? Not entirely sure
```suggestion
    if (input.userId === ctx.user.id && !ctx.user.isModerator) {
```